test/middleware.test.js

```diff
@@ -91,7 +91,7 @@ describe('loopback:middleware generator', function() {
         var newSources = Object.keys(
           readMiddlewaresJsonSync('server')['routes:after']);
         var expectedSources = builtinSources.concat(['my-middleware-4']);
-        expect(newSources).to.have.members(expectedSources);
+        expect(expectedSources).to.have.include.members(newSources);
```

Comment:
You should fix L88-L89 and set `builtinSource` to the content of `routes:after` phase instead of current `routes` phase. After that, `expectedSources` should contain both default `loopback#rest` and added `my-middleware-4`.